amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ProcessTask.java

```diff
@@ -120,8 +119,7 @@ public TaskResult call() {
          */
         final MetricsScope appScope = MetricsUtil.createMetricsWithOperation(metricsFactory, APPLICATION_TRACKER_OPERATION);
         final MetricsScope shardScope = MetricsUtil.createMetricsWithOperation(metricsFactory, PROCESS_TASK_OPERATION);
-        shardInfo.streamIdentifierSerOpt()
-                .ifPresent(streamId -> MetricsUtil.addStreamId(shardScope, StreamIdentifier.multiStreamInstance(streamId)));
+        MetricsUtil.addStreamId(shardScope, shardInfo.streamConfig().streamIdentifier());
```

Comment:
Okay we dont want to do this by default, for single stream application, this can change dimension and may break public contract on the metrics exposed by KCL.

Seems like then we need some method from streamIdentifier, you had isMultiStreamInstance. We could use it, but I dint want to expose that method based on the comment I left there. Lets think if there is another way.